packages/fuel-gauge/src/funding-transaction.test.ts

```diff
@@ -232,4 +236,109 @@ describe(__filename, () => {
 
     expect(getResourcesToSpend).toHaveBeenCalledTimes(2);
   });
+
+  it('should ensure a partially funded Transaction will require only missing funds', async () => {
+    const receiver = Wallet.generate({ provider });
+    const wallet1 = Wallet.generate({ provider });
+    const wallet2 = Wallet.generate({ provider });
+
+    const totalInBaseAsset = 15_000;
+    const totalInAssetA = 20_000;
+    const partiallyInAssetA = totalInAssetA / 2;
+
+    // wallet1 does not have enough funds to cover total required in assetA and
+    await seedTestWallet(wallet1, [
+      [totalInBaseAsset, baseAssetId],
+      [partiallyInAssetA, assetA],
+    ]);
+
+    /**
+     * The wallet2 will have the missing amount to cover the total required in assetA and
+     * will not have enough funds to cover the fee.
+     */
+    await seedTestWallet(wallet2, [[partiallyInAssetA, assetA]]);
+
+    let transactionRequest = new ScriptTransactionRequest();
+
+    transactionRequest.addCoinOutput(receiver.address, totalInAssetA, assetA);
+
+    const txCost = await provider.getTransactionCost(transactionRequest);
+
+    transactionRequest.gasLimit = txCost.gasUsed;
+    transactionRequest.maxFee = txCost.maxFee;
+
+    const partiallyResources = await wallet1.getResourcesToSpend([
+      [partiallyInAssetA, assetA],
+      [totalInBaseAsset, baseAssetId],
+    ]);
+
+    const baseAssetResource = partiallyResources.find((r) => r.assetId === baseAssetId);
+    const assetAResource = partiallyResources.find((r) => r.assetId === assetA);
+
+    // expect to have the correct amount of resources, not enough to cover the TX needs
+    expect(baseAssetResource?.amount.toString()).toBe(totalInBaseAsset.toString());
+    expect(assetAResource?.amount.toString()).toBe(partiallyInAssetA.toString());
+
+    transactionRequest.addResources(partiallyResources);
```

Comment:
Maybe I am misinterpreting the issue but shouldn't this test essentially ensure that the `fund` method is only called  when resources if they are missing? Or perhaps the method should reject an attempt to add resources if they are present?

Reply:
@maschad No, the issue is to validate that the `Account.fund` method will not try to fetch and add more funds to a TX that is already funded.